components/infobox/wikis/starcraft2/infobox_person_custom_shared.lua

```diff
@@ -0,0 +1,176 @@
+		or Variables.varDefault('disable_SMW_storage', 'false') == 'true'
+		or mw.title.getCurrentTitle().nsText ~= ''
+	then
+		Variables.varDefine('disable_SMW_storage', 'true')
+		return false
+	end
+	return true
+end
+
+function CustomPerson.nameDisplay()
+	CustomPerson.getRaceData(_args.race or 'unknown')
+	local raceIcon = RaceIcon({'alt_' .. _raceData.race})
+	local name = _args.id or _PAGENAME
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function CustomPerson.getRaceData(race)
+	race = string.lower(race)
+	race = CleanRace[race] or race
+	local display = CustomPerson.raceDisplayLookupTable[race]
+	if not display and race ~= 'unknown' then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	end
+
+	_raceData = {
+		race = race,
+		faction = _FACTION1[race] or '',
+		faction2 = _FACTION2[race] or '',
+		display = display
+	}
+
+	return display
+end
+
+function CustomPerson.adjustLPDB(_, lpdbData)
+	local extradata = lpdbData.extradata or {}
+	extradata.race = _raceData.race
+	extradata.faction = _raceData.faction
+	extradata.faction2 = _raceData.faction2
+	extradata.lc_id = string.lower(_PAGENAME)
+	extradata.teamname = _args.team
+	extradata.role = _args.role
+	extradata.role2 = _args.role2
+	extradata.militaryservice = _militaryStore
+	extradata.activeplayer = (not _statusStore) and Variables.varDefault('isActive', '') or ''
+
+	if Variables.varDefault('racecount') then
+		extradata.racehistorical = true
+		extradata.factionhistorical = true
+	end
+
+	lpdbData.extradata = extradata
+
+	return lpdbData
+end
+
+function CustomPerson.military(military)
+	if military and military ~= 'false' then
+		local display = military
+		military = string.lower(military)
+		local militaryCategory = ''
+		if String.contains(military, 'starting') or String.contains(military, 'pending') then
+			militaryCategory = '[[Category:Persons waiting for Military Duty]]'
+			_militaryStore = 'pending'
+		elseif
+			String.contains(military, 'ending') or String.contains(military, 'started')
+			or String.contains(military, 'ongoing')
+		then
+			militaryCategory = '[[Category:Persons on Military Duty]]'
+			_militaryStore = 'ongoing'
+		elseif String.contains(military, 'fulfilled') then
+			militaryCategory = '[[Category:Persons expleted Military Duty]]'
+			_militaryStore = 'fulfilled'
+		elseif String.contains(military, 'exempted') then
+			militaryCategory = '[[Category:Persons exempted from Military Duty]]'
+			_militaryStore = 'exempted'
+		end
+
+		return display .. militaryCategory
+	end
+end
+
+function CustomPerson.getStatusToStore()
+	if _args.death_date then
+		_statusStore = 'Deceased'
+	elseif _args.retired then
+		_statusStore = 'Retired'
+	elseif
+		(not Logic.readBool(_args.isplayer)) and
+		string.lower(_args.role or _args.defaultPersonType) ~= 'player'
+	then
+		_statusStore = 'not player'
```

Comment:
Is there a risk that a none-player who retires is believed to be a player by another module (as they are retired, we won't store "not player" in the status)?

Reply:
retired casters/players/etc are treated the same mostly, so not really (it is handled like that in the current implementation too btw.)

Reply:
Ok.

FYI you have a bug in the code for this page https://liquipedia.net/starcraft2/Players_(All)#USA. All casters (even active ones) end up under retired dropdown (eg see Ukraine -> The TheFirstBee). Also the USA is just broken

Reply:
ah yeah forgot to switch usa to united states on that page
the other one was a typo in the portal player module, fixed now